plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java

```diff
@@ -99,11 +112,55 @@ protected Composite createDialogArea(Composite parent) {
         sqlGenerator.setShowFullDdl(getDialogBoundsSettings().get(DBPScriptObject.OPTION_INCLUDE_NESTED_OBJECTS) != null &&
                 getDialogBoundsSettings().getBoolean(DBPScriptObject.OPTION_INCLUDE_NESTED_OBJECTS));
 
-        UIUtils.runInUI(sqlGenerator);
-        Object sql = sqlGenerator.getResult();
-        if (sql != null) {
-            setSQLText(CommonUtils.toString(sql));
-        }
+        generateDDLJob = new Job("Generating DDL") {
+            @Override
+            protected IStatus run(IProgressMonitor monitor) {
+                try {
+                    sqlGenerator.run(new DefaultProgressMonitor(monitor));
+                    return Status.OK_STATUS;
+                } catch (InvocationTargetException e) {
+                    log.error(e);
+                    return Status.error("Error running while DDL generation", e);
+                } catch (InterruptedException ignore) {
+                    return Status.CANCEL_STATUS;
+                }
+            }
+        };
+        generateDDLJob.addJobChangeListener(new JobChangeAdapter() {
+            @Override
+            public void running(IJobChangeEvent event) {
+                if (SQLGeneratorDialog.this.getShell() != null && !SQLGeneratorDialog.this.getShell().isDisposed()) {
+                    UIUtils.syncExec(() -> {
+                        setSQLText("Loading DDL...");
+                        updateSQL();
+                        Button button = getButton(IDialogConstants.DETAILS_ID);
+                        if (button != null && !button.isDisposed()) {
```

Comment:
What for? Where is this button?

Reply:
Copy button

Reply:
No reason to copy if DDL is not yet ready

Reply:
ok.